crates/transaction-pool/src/validate.rs

```diff
@@ -18,11 +24,11 @@ pub enum TransactionValidationOutcome<T: PoolTransaction> {
         /// Current nonce of the sender.
         state_nonce: u64,
         /// Validated transaction.
-        transaction: T,
+        transaction: Arc<T>,
```

Comment:
why is wrapping in Arce now necessary? 

this is not great because `transaction` will be wrapped in an Arc anyways but encapsulated in ValidPoolTransaction